src/land/madis_snow2ioda.py

```diff
@@ -0,0 +1,160 @@
+    ("dateTime", "long")
+]
+
+obsvars = {
+    'snow_depth': 'totalSnowDepth',
+}
+
+AttrData = {
+}
+
+DimDict = {
+}
+
+VarDims = {
+    'totalSnowDepth': ['Location'],
+}
+
+iso8601_string = 'seconds since 1970-01-01T00:00:00Z'
+epoch = datetime.fromisoformat(iso8601_string[14:-1])
+
+class madis(object):
+    def __init__(self, filename, mask):
+        self.filename = filename
+        self.mask = mask
+        self.varDict = defaultdict(lambda: defaultdict(dict))
+        self.outdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
+        self.varAttrs = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
+        self._read()
+
+    # Open input file and read relevant info
+    def _read(self):
+        # set up variable names for IODA
+        for iodavar in ['totalSnowDepth']:
+            self.varDict[iodavar]['valKey'] = iodavar, iconv.OvalName()
+            self.varDict[iodavar]['errKey'] = iodavar, iconv.OerrName()
+            self.varDict[iodavar]['qcKey'] = iodavar, iconv.OqcName()
+            self.varAttrs[iodavar, iconv.OvalName()]['coordinates'] = 'longitude latitude'
+            self.varAttrs[iodavar, iconv.OerrName()]['coordinates'] = 'longitude latitude'
+            self.varAttrs[iodavar, iconv.OqcName()]['coordinates'] = 'longitude latitude'
+            self.varAttrs[iodavar, iconv.OvalName()]['units'] = 'mm'
+            self.varAttrs[iodavar, iconv.OerrName()]['units'] = 'mm'
+
+        # open input file name
+        ncd = nc.Dataset(self.filename, 'r')
+        vals = ncd.variables['snowDepth'][:]
+        _FillValue = ncd.variables['snowDepth'].getncattr('_FillValue')
+
+        lats = ncd.variables['latitude'][:]
+        lons = ncd.variables['longitude'][:]
+        elvs = ncd.variables['elevation'][:]
+        qflg = ncd.variables['snowDepthQCR'][:]
+        obst = ncd.variables['observationTime'][:]
+        sids = nc.chartostring(ncd.variables['stationId'][:,:])
+    
+        sites = np.empty_like(vals, dtype=object)
+        for i in range(len(vals[:])):
+            sites[i] = sids[i]
+        times = np.empty_like(vals, dtype=np.int64)
+        times = obst
+
+        vals = vals.astype('float32')
+        lats = lats.astype('float32')
+        lons = lons.astype('float32')
+        elvs = elvs.astype('float32')
+        errs = 0.0*vals
+        errs = errs.astype('float32')
+        qflg = qflg.astype('int32')
+
+        # use maskout options
+        if self.mask == "maskout":
+
+            with np.errstate(invalid='ignore'):
+                mask = (vals >= 0.0) & (vals < 2000.0)
+            vals = vals[mask]
+            errs = errs[mask]
+            qflg = qflg[mask]
+            lons = lons[mask]
+            lats = lats[mask]
+            elvs = elvs[mask]
+            sites = sites[mask]
+            times = times[mask]
+
+        for i in range(len(vals)):
+            if vals[i] >= 0.0:
+                errs[i] = 40.0
+            else:
+                errs[i] = _FillValue
```

Comment:
The fill values are not getting set properly on the totalSnowDepth in the ObsValue and ObsError groups in the output ioda file. Here's an excerpt of running ncdump on the output file from the new test:
```
group: ObsError {
  variables:
        float totalSnowDepth(Location) ;
                totalSnowDepth:_FillValue = 9.96921e+36f ;
                string totalSnowDepth:coordinates = "longitude latitude" ;
                string totalSnowDepth:units = "mm" ;
  data:

   totalSnowDepth = 40, 40, 40, 40, 40, 40, 40, 40, 40, 40, 3.402823e+38, 
      3.402823e+38, 40, 40, 40, 40, 40, 40 ;
  } // group ObsError

group: ObsValue {
  variables:
        float totalSnowDepth(Location) ;
                totalSnowDepth:_FillValue = 9.96921e+36f ;
                string totalSnowDepth:coordinates = "longitude latitude" ;
                string totalSnowDepth:units = "mm" ;
  data:

   totalSnowDepth = 0, 571.5, 584.2, 635, 4826, 573.024, 459.994, 93.98, 0, 
      106.68, 3.402823e+38, 3.402823e+38, 711.2, 457.2, 939.8, 863.6, 25.4, 
      76.2 ;
  } // group ObsValue
```
Note that the variable attribute named "_FillValue" is set to 9.96921e+36f (which is the netcdf default for a float32) while the elements in the data are set to 3.402823e+38 (which is the value that was read from the input file).

This can be fixed by adding the attribute "_FillValue", set to 3.402823e+38, to the VarAttrs dictionary in the loop below on lines 120-123. You will need to add lines like these to the loop:
```
self.varAttrs[self.varDict[iodavar]['valKey']]['_FillValue'] = _FillValue
self.varAttrs[self.varDict[iodavar]['errKey']]['_FillValue'] = _FillValue
```
You will know when the attributes are set correctly when ncdump prints "_" instead of 3.402823e+38 in the data locations where the missing values are located.

Reply:
This has been fixed as suggested.